src/main/java/org/datadog/jenkins/plugins/datadog/traces/DatadogTracePipelineLogic.java

```diff
@@ -46,15 +45,13 @@ public JSONObject toJson(PipelineStepData flowNode, Run<?, ?> run) throws IOExce
     public TraceSpan toSpan(PipelineStepData current, Run<?, ?> run) throws IOException, InterruptedException {
         BuildData buildData = new BuildData(run, DatadogUtilities.getTaskListener(run));
 
-        // If the root span has propagated queue time, we need to adjust all startTime and endTime from Jenkins pipelines spans
-        // because this time will be subtracted in the root span. See DatadogTraceBuildLogic#finishBuildTrace method.
-        final long propagatedMillisInQueue = Math.max(buildData.getPropagatedMillisInQueue(-1L), 0);
-        final long fixedStartTimeNanos = TimeUnit.MICROSECONDS.toNanos(current.getStartTimeMicros() - TimeUnit.MILLISECONDS.toMicros(propagatedMillisInQueue));
-        final long fixedEndTimeNanos = TimeUnit.MICROSECONDS.toNanos(current.getEndTimeMicros() - TimeUnit.MILLISECONDS.toMicros(propagatedMillisInQueue));
+        final long startTimeNanos = TimeUnit.MILLISECONDS.toNanos(current.getStartTimeMillis());
+        final long endTimeNanos = TimeUnit.MILLISECONDS.toNanos(current.getEndTimeMillis());
 
         // At this point, the current node is traceable.
         final TraceSpan.TraceSpanContext spanContext = new TraceSpan.TraceSpanContext(current.getTraceId(), current.getParentSpanId(), current.getSpanId());
-        final TraceSpan span = new TraceSpan(buildOperationName(current), fixedStartTimeNanos + current.getNanosInQueue(), spanContext);
+        final TraceSpan span = new TraceSpan(buildOperationName(current), startTimeNanos + current.getQueueTimeMillis(), spanContext);
```

Comment:
Are we removing the queue time twice? We're doing the same [in this part of the code](https://github.com/jenkinsci/datadog-plugin/blob/d2537c85e514e669736c53aed5a75c587d5740e5/src/main/java/org/datadog/jenkins/plugins/datadog/model/PipelineStepData.java#L173) and [we're not doing it for webhooks protocol.](https://github.com/jenkinsci/datadog-plugin/blob/d2537c85e514e669736c53aed5a75c587d5740e5/src/main/java/org/datadog/jenkins/plugins/datadog/traces/DatadogWebhookPipelineLogic.java#L55)

Reply:
Good catch, fixed!